### DISCUSSION TITLE
Creating Objects with ID

### DISCUSSION THREAD

**Discussion by sabrikaragonen** [fastapi/fastapi#7657] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Actually, it more sounds like pydantic issue, but since I wonder out of pydantic solution methods, I'm asking this question here.

I want to construct an object only with ID, and then populate other fields based on ID.
I tried 2 different approaches, validator and also post init. Neither did work, so.

Common code:
```python
from pydantic import BaseModel, validator

obj_list = [
    {'id': 1, 'name': 'a', 'desc': 'desc1'},
    {'id': 2, 'name': 'b', 'desc': 'desc2'},
    {'id': 3, 'name': 'c', 'desc': 'desc3'}
]
```

Solution 1:  
```python
class Obj(BaseModel):
    id: int
    name: str = None
    desc: str = None
        
    @validator('id')
    def validate_exists(cls, v):
        items = [h for h in obj_list if h['id'] == v]
        if len(items) == 0:
            raise ValueError('name doesnt exist')
        return v
    
    def __post_init__(self):
        item = [h for h in obj_list if h['name'] == v][0]
        self.name = item['name']
        self.desc = item['desc']
        
Obj(id=1)
# <Obj id=1 name=None desc=None>
```

Solution 2: 
```python
class Obj(BaseModel):
    name: str = None
    desc: str = None
    id: int
        
    @validator('id')
    def validate_exists(cls, v, values):
        items = [h for h in obj_list if h['id'] == v]
        if len(items) == 0:
            raise ValueError('name doesnt exist')
        item = items[0]
        values['name'] = item['name']
        values['desc'] = item['desc']
        return v
        
Obj(id=1)
# <Obj name=None desc=None id=1>
```

I feel like it's doable. I read both Pydantic's and FastApi's documentation, but couldn't find anything relevant to this.
So, how can I make it happen?

**Comment by trim21** [comment]
It's possible to validate multiple field and set their values in one validator, there are more details in pydantic docs

**Comment by sabrikaragonen** [comment, marked as answer]
I solved it using root_validators from pydantic. Thanks!

Solution is here:
https://stackoverflow.com/a/60744789/3733295

**Comment by tiangolo** [comment]
Thanks for the help here @Trim21 

Thanks for reporting back and closing the issue @sabrikaragonen :+1:
